src/articles/articles.controller.ts

```diff
@@ -18,16 +21,20 @@ export class ArticlesController {
   }
 
   @Get(':id')
+  @Authz(new Permission('read', 'article'))
   findOne(@Param('id') id: string) {
-    return this.articlesService.findOne(+id);
+    return this.articlesService.findOne(id);
   }
 
   @Put(':id')
+  // IsAuthor is additional condition
+  @Authz(new Permission('update', 'article'), IsAuthor)
```

Comment:
suggestions for other api, after some search into the rbac papers and best practices. 
- article not a subject 
- and literature calls action operation. i prefer the term action however.
- we shouldn't put a permission in the policy checker, because the permission is *to be checked*

API suggestion:
```
@AuthAction('update-action', IsAuthor)
```
and typesafe version, with an action definition per module:
```
@AuthAction(UpdateAction, IsAuthor)
```